### DISCUSSION TITLE
Rename to SpecSharp?

### DISCUSSION THREAD

**Discussion by markdav-is** [reqnroll/Reqnroll#194] [Ideas]
I respect the big lift it is to name a product.  We've been switching from SpecFlow to Reqnroll this year.  Everyone stumbles over the name and we still refer to it as SpecFlow for the most part.  On a whim I looked to see if SpecSharp.com was available today and it is.  I'm going to grab that domain with the sole intention of giving it to this project and helping with a rename if the core team feels that would be a better name.

**Comment by gasparnagy** [comment]
Thank you @markdav-is for the feedback. 

We have spent a lot of time with the naming. It is always hard. We have also considered many "Spec*" product names, including SpecSharp, but finally decided to not have a "Spec" prefix for several reasons:
- `SpecFlow` is a registered trademark and I wanted to avoid any legal case where I would need to prove that the new name is not misleadingly similar to SpecFlow. I'm sure we would have won at the end, but the last thing I needed was to deal with lawyers (I'm a tech person).
- We created many tools with the "Spec" prefix at the TechTalk times, and they were confusing people, so with a name, like SpecSharp, we would just get back this confusion, that I did not wanted (We have limited time for support - it is all free work).
- I have my own company (Spec Solutions) that also have a "Spec*" commercial product, called SpecSync. (Also caused quite some confusion, so that was part of the learning.) I wanted to make the Reqnroll project as an independent initiative. With a name, like SpecSharp, it could have sounded like that I make a "promo-open-source" project to get attention to my product or to my company. So getting sponsors would have been harder that way and I did not feel that ethical.

At the end, it was a decision. I'm not sure it was the right one, but at least it was well thought. And I also had to realize that "renaming a product" is a horrible effort, much more than I expected (I have spent around 40 days). So I don't think it is economical to make another rename at this time, sorry. So my suggestion would be not to reopen the naming topic, but admit the situation we have now and concentrate to feature improvements.

It took me some time to get used to the new name, but after a couple of month of working, now I automatically say Reqnroll when I think of it. So it will take time (I expect the full migration in the community to last for 2 years), but it will work.

**Reply by Digbigpig** [reply]
Just wanted to point out (although maybe not necessary) that the website has this copy, we may want to reword that if the name is set in stone for the time being.

> I also considered calling it ... although this is something we can reconsider later.

**Reply by gasparnagy** [reply]
@Digbigpig Thx for pointing that out. The form of Cucumber collaboration is a tough question anyway with a lot of side-questions. But the Cucumber project has not finished moving out to a foundation, so we cannot even start that discussion yet. Let's come back to that once that is done. (But nevertheless the more time passes the harder it will to make name changes.)
